Test failure in test_trainer.TestTrainer

https://ci.appveyor.com/project/pfnet/chainer/build/3973

```
FAIL: test_elapsed_time (test_trainer.TestTrainer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\chainer\tests\chainer_tests\training_tests\test_trainer.py", line 65, in test_elapsed_time
    self.assertGreater(self.trainer.elapsed_time, 0)
AssertionError: 0.0 not greater than 0
======================================================================
FAIL: test_elapsed_time_serialization (test_trainer.TestTrainer)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\chainer\tests\chainer_tests\training_tests\test_trainer.py", line 75, in test_elapsed_time_serialization
    self.assertGreater(new_trainer.elapsed_time, serialized_time)
AssertionError: 0.0 not greater than 0.0
```

--- a/chainer/testing/training.py
+++ b/chainer/testing/training.py
@@ -3,8 +3,7 @@ import mock
 from chainer import training
 
 
-def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration'),
-                                  update_core=None):
+def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration')):
     """Returns a :class:`~chainer.training.Trainer` object with mock updater.
 
     The returned trainer can be used for testing the trainer itself and the
@@ -14,7 +13,6 @@ def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration'),
 
     Args:
         stop_trigger: Stop trigger of the trainer.
-        update_core: A function object to mock ``updater.update_core`` method.
 
     Returns:
         Trainer object with a mock updater.
@@ -24,7 +22,6 @@ def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration'),
     updater.get_all_optimizers.return_value = {}
     updater.iteration = 0
     updater.epoch_detail = 1
-    updater.update_core = update_core or (lambda: None)
 
     def update():
         updater.update_core()

--- a/chainer/testing/training.py
+++ b/chainer/testing/training.py
@@ -3,7 +3,8 @@ import mock
 from chainer import training
 
 
-def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration')):
+def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration'),
+                                  update_core=None):
     """Returns a :class:`~chainer.training.Trainer` object with mock updater.
 
     The returned trainer can be used for testing the trainer itself and the
@@ -13,6 +14,7 @@ def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration')):
 
     Args:
         stop_trigger: Stop trigger of the trainer.
+        update_core: A function object to mock ``updater.update_core`` method.
 
     Returns:
         Trainer object with a mock updater.
@@ -22,8 +24,10 @@ def get_trainer_with_mock_updater(stop_trigger=(10, 'iteration')):
     updater.get_all_optimizers.return_value = {}
     updater.iteration = 0
     updater.epoch_detail = 1
+    updater.update_core = update_core or (lambda: None)
 
     def update():
+        updater.update_core()
         updater.iteration += 1
 
     updater.update = update

--- a/tests/chainer_tests/training_tests/test_trainer.py
+++ b/tests/chainer_tests/training_tests/test_trainer.py
@@ -54,9 +54,13 @@ class DummyClass(object):
 class TestTrainer(unittest.TestCase):
 
     def setUp(self):
-        self.trainer = testing.get_trainer_with_mock_updater(
-            update_core=lambda: time.sleep(0.001))
+        self.trainer = self._create_mock_trainer(10)
         self.trainer.is_initialized = False
+
+    def _create_mock_trainer(self, iterations):
+        return testing.get_trainer_with_mock_updater(
+            (iterations, 'iteration'),
+            update_core=lambda: time.sleep(0.001))
 
     def test_elapsed_time(self):
         with self.assertRaises(RuntimeError):
@@ -70,7 +74,7 @@ class TestTrainer(unittest.TestCase):
         self.trainer.run()
         serialized_time = self.trainer.elapsed_time
 
-        new_trainer = testing.get_trainer_with_mock_updater((20, 'iteration'))
+        new_trainer = self._create_mock_trainer(5)
         testing.save_and_load_npz(self.trainer, new_trainer)
 
         new_trainer.run()

--- a/tests/chainer_tests/training_tests/test_trainer.py
+++ b/tests/chainer_tests/training_tests/test_trainer.py
@@ -1,3 +1,4 @@
+import time
 import unittest
 
 from chainer import testing
@@ -53,7 +54,8 @@ class DummyClass(object):
 class TestTrainer(unittest.TestCase):
 
     def setUp(self):
-        self.trainer = testing.get_trainer_with_mock_updater()
+        self.trainer = testing.get_trainer_with_mock_updater(
+            update_core=lambda: time.sleep(0.001))
         self.trainer.is_initialized = False
 
     def test_elapsed_time(self):

--- a/tests/chainer_tests/training_tests/test_trainer.py
+++ b/tests/chainer_tests/training_tests/test_trainer.py
@@ -58,9 +58,10 @@ class TestTrainer(unittest.TestCase):
         self.trainer.is_initialized = False
 
     def _create_mock_trainer(self, iterations):
-        return testing.get_trainer_with_mock_updater(
-            (iterations, 'iteration'),
-            update_core=lambda: time.sleep(0.001))
+        trainer = testing.get_trainer_with_mock_updater(
+            (iterations, 'iteration'))
+        trainer.updater.update_core = lambda: time.sleep(0.001)
+        return trainer
 
     def test_elapsed_time(self):
         with self.assertRaises(RuntimeError):

